kedro/io/core.py

```diff
@@ -290,6 +290,18 @@ def _describe(self) -> Dict[str, Any]:
             "it must implement the `_describe` method".format(self.__class__.__name__)
         )
 
+    def set_max_loads(self, max_loads: int):
```

Comment:
Shouldn't `max_loads` default to `None` if we allow it to not be specified?
(It seems I missed it in the initial implementation of `create_default_data_set`)

Reply:
Not sure what you mean? Why would you call this method without specifying the `max_loads`?

Reply:
In the original implementation if `max_loads is None`, it allows unlimited number of loads